### PR TITLE
New Task TTL behavior once the result is available

### PR DESCRIPTION
**Pull request by yadudoc**
The Task TTL currently is set to 2 weeks, which makes sense when tasks might take a while to execute. However, we shouldn't need to hold tasks in REDIS when the result is available for longer than an hour. The docs have been updated separately to list the TTL after a task is complete to be 30 mins.

Ideally the minor change to the Task class here should go into funcx-common. For now this will do.

@joshbryan-globus, the configurability comes from the entrypoint expecting an env var `RESULT_TTL_SECONDS` which can be configured from helm, but that last bit needs a separate PR.

### CODE REVIEW

File: funcx_forwarder/forwarder.py [hunk 1]
```diff
@@ -523,6 +528,7 @@ def handle_results(self):
                 task.status = TaskState.FAILED
                 task.exception = message['exception']
                 task.completion_time = time.time()
+            task._set_expire(self.result_ttl)
```
Comment (joshbryan-globus) [inline funcx_forwarder/forwarder.py:531]: Didn't this change to `set_expire` rather than `_set_expire`?

**Review by sirosen — APPROVED** [review]
This looks good.

As you say, we want to add a RedisTask class or base class to the common lib to handle this and other activities. But for now, the task handling is still a bit fragmented.

I see you added this as a CLI opt. FWIW, I'd be happy with just checking for an env var with os.getenv, in future cases like this. It might save you a couple of steps tweaking the entrypoint and arg parser. But that's no big deal either way.

**Comment by yadudoc** [comment]
Let's put merging this one on. I'm seeing the call to `set_expire` go through but the TTL on the keys in redis does not seem to get updated.

**Comment by yadudoc** [comment]
This is tested and all good now.

**Pull request merged**